policy/volatile_ttl.cpp

```diff
@@ -0,0 +1,170 @@
+    {
+        std::vector<ttl_item> ttl_items;
+        auto char_reader{Json::CharReaderBuilder().newCharReader()};
+        std::transform(our_ttls.begin(), our_ttls.end(), std::back_inserter(ttl_items),
+                [&](const auto& ttl)->ttl_item
+                {
+
+                    Json::Value root;
+                    std::string err;
+                    if (char_reader->parse(ttl.c_str(), ttl.c_str() + ttl.size(), &root, &err))
+                    {
+                        const auto key{root["key"].asString()};
+                        const auto expire_key{this->generate_expire_key(db_uuid, key)};
+                        const auto opt_expire{storage->read(TTL_UUID, expire_key)};
+                        const auto opt_value{storage->read(db_uuid, key)};
+
+                        if (!opt_value.has_value())
+                        {
+                            LOG (warning) << "item with key:[" << key << "] does not exist in db: [" <<  db_uuid << "]";
+
+                            return ttl_item{"", 0 , 0};
+                        }
+
+                        if (!opt_expire.has_value())
+                        {
+                            LOG (warning) << "TTL item with key:[" << expire_key << "] does not exist";
+
+                            return ttl_item{"", 0, 0};
+                        }
+
+                        LOG (info) << "Found an item to evict: [" << key << "]" ;
+
+                        return ttl_item{
+                            key
+                            , opt_value.value().size()
+                            , boost::lexical_cast<uint64_t>(opt_expire.value())
+                        };
+                    }
+                    LOG(warning) << "Unable to parse TTL for Volatile TTL eviction: " << err;
+
+                    return ttl_item{"", 0 , 0};
+                });
+
+        return ttl_items;
+    }
+
+
+    std::vector<std::string>
+    volatile_ttl::filter_ttls(const std::vector<std::string>& ttls, const bzn::uuid_t& db_uuid, const bzn::key_t& ignore_key)
+    {
+        std::vector<std::string> filtered_ttls;
+        auto char_reader{Json::CharReaderBuilder().newCharReader()};
+        std::copy_if(
+                ttls.begin(), ttls.end(), std::back_inserter(filtered_ttls)
+                , [&char_reader, &db_uuid, &ignore_key](const auto& t)
+                {
+                    Json::Value root;
+                    std::string err;
+                    if (char_reader->parse(t.c_str(), t.c_str() + t.size(), &root, &err))
+                    {
+                        bool s{(root["key"] != ignore_key) && (root["uuid"].asString() == db_uuid)};
+
+                        return s;
+                    }
+                    LOG(warning) << "Unable to parse TTL for Volatile TTL eviction: " << err;
+
+                    return false;
+                });
+        delete char_reader;
```

Comment:
use unique_ptr

Reply:
Made the change, also found the same bad pattern elsewhere.